File: 2566. Maximum Difference by Remapping a Digit/Solution.java
class Solution {
    public int minMaxDifference(int num) {
        String s = Integer.toString(num);
        char c = s.charAt(0), ch = '9';
        int l = s.length(), j = 0;
        while (j < l) {
            if (s.charAt(j) < '9') {
                ch = s.charAt(j);
                break;
            }
            j++;
        }
        StringBuilder high = new StringBuilder(), low = new StringBuilder();
        for (int i = 0; i < l; i++) {
            char curr = s.charAt(i);
            if (curr == c) {
                low.append('0');
            } else {
                low.append(curr);
            }
            if (curr == ch) {
                high.append('9');
            } else {
                high.append(curr);
            }
        }
        return Integer.parseInt(high.toString()) - Integer.parseInt(low.toString());
    }
}

File: 2566. Maximum Difference by Remapping a Digit/Solution.py
class Solution:
    def minMaxDifference(self, num: int) -> int:
        s = str(num)
        c, ch = s[0], '9'
        for cs in s:
            if cs != '9':
                ch = cs
                break
        high, low = "", ""
        for cs in s:
            if cs == c:
                low += "0"
            else:
                low += cs
            if cs == ch:
                high += "9"
            else:
                high += cs
        return int(high) - int(low)